### DISCUSSION TITLE
Apps Performance

### DISCUSSION THREAD

**Discussion by SkullFace141** [engageub/InternetIncome#33] [General]
Apps Performance:
EarnAPP: terrible 200mb per day
Honeygain: Good ~30GB per day
PawnsAPP: I am not using it
ProxyRack: terrible and barely working 0.1GB per day
PacketStream: Good 8GB per day
Traffimonitizer: working with low traffic 3GB per day
Repocket: Working but the network is saturated few cents per day
Ebuscher: I am not using it
ProxyLite: I am not using it
Mysterium: Testing one node

App Performance on my manual setup:
EarnAPP: Fair 6.5GB per day
Honeygain: Good ~30GB per day
PawnsAPP: I am not using it
ProxyRack: Good 9GB per day
PacketStream: Good 8GB per day
Traffimonitizer: Good 8GB per day
Repocket: Working but the network is saturated few cents per day
Ebuscher: I am not using it
ProxyLite: I am not using it
Mysterium: I am not using it

Conclusion:
There are issues with EarnAPP, ProxyRack and Traffmonitizer for some reason, i have been using the same proxies on the same machine, RAM is ~50% used CPU is ~55% used, Internet Speed 1GBPS shared, number of proxies 100

**Comment by engageub** [comment]
Hi,
Thank you for providing the stats. Could you please let me know if you have used the proxies on same location.
Also earnapp checks the CPU and memory. You may have a look at the logs if it shows that CPU is not idle.
It was working good as other users confirmed it and it had good traffic. 
Also, I am assuming that your PC was not rebooted and the instances were running since the start of the script. 
If you rebooted the PC while it was already running DNS may not work if your proxy does not support UDP. 
I would like to know how you tested manually with 100 proxies. Do you have 100 instances?

Thank you

**Comment by SkullFace141** [comment]
same proxies, same machine and the machine was not rebooted since the scrip[t started.
the manual setup has 100 LXC containers, each container has 1 proxy and 1 instance of each app

**Reply by TradeBW** [reply]
Hello, did you use proxy via export= or env or did you use some proxying tool on lxc ? Thanks in advance

**Comment by engageub** [comment]
Thank you for the data. You may have to check with the respective container app owners who are maintaining this. The script has been tested for proxy use and deployed. 
It cannot be confirmed on what is going on unless the issue has been identified. There might be many reasons why this is occurring with earnapp application. The traffic is very low when compared to the manual setup.  Users claim  they get about 2GB traffic per day from 10 IP for earnapp in US region.  
Looking at the honeygain data, it is consistent across both docker and LXC. The script has no impact and it only proxies the containers using the available applications.  If it is not working as expected you may raise an issue.
Also how is the CPU and memory utilization when compared to LXC and docker containers. Do 100 instances of LXC containers use the same CPU and memory as the docker containers for the same number of apps used.

Thank you

**Comment by SkullFace141** [comment]
LXC use more resources

**Comment by engageub** [comment]
Thank you for the confirmation. So docker is performing better when compared to LXC but earnings are not as good as LXC containers according to your system analysis.
If you have good CPU and memory, you can install salad and see your earnings. Some of the countries it pays about $1 per day. 

Thank you

**Reply by UncleJack555** [reply]
may i have a question? how to run a app like honeygain in LXC container? it need to go through docker?

**Reply by engageub** [reply]
@UncleJack555 They are different platforms. LXC containers do not require docker to work.
@SkullFace141 , Could you please provide instructions on how to run LXC containers.
Thank you

**Reply by UncleJack555** [reply]
btw, how can i get PACKETSTREAM_CID= token...? It's required in properties.conf..

**Reply by engageub** [reply]
@UncleJack555 , It will be present in your dashboard-downloads section.
1. Click on "Download" from the packetstream dashboard
2. Then click on "Looking for the Mac or Linux app? "
3. Scroll down to the docker command line where you will see your CID number. Search for keyword CID in the page and you will find your CID number in the format below. Copy the CID number which is usually 4 characters and paste in properties.conf file. 
```
CID=<YOUR_CID_NUMBER>
```

**Reply by UncleJack555** [reply]
@engageub thanks a lot dude :D. U're amazing author, fastest respond person I ever seen. Your script worked so well through tun2socks (im wonder can i manual dat images at each single container?). However, im still cant using it on earnapp, honeygain(sometimes it's worked..) and IProyal.. I guess cuz my proxies just too bad T.T.. But they're working hard on packetstream rite now xD dont understand why, lul

![skd-not-earn](https://github.com/engageub/InternetIncome/assets/143498790/aa00a0ae-b251-4f5c-a65b-8eda224700f5)

**Reply by engageub** [reply]
@UncleJack555 
Thank you for your feedback. I did not understand what you meant by "(im wonder can i manual dat images at each single container?)."
Please note that you have to get good residential proxies to earn for which you are responsible. 
If you have any issues, you can enable logs from properties.conf and look into the container logs.

Thank you

**Reply by UncleJack555** [reply]
@engageub can I use your script to play earnapp on VPS linux? Im running 40 containers (with earnapp only).. my real PC RAM's out of space now..T.T

**Reply by engageub** [reply]
@UncleJack555 Yes, you can run as many containers as you want as long as the resources are available.

Thank you

**Reply by UncleJack555** [reply]
great. thanks so much for confirmed xD!

**Comment by engageub** [comment]
Can you please provide an update if there are any additional tasks remaining in this matter?

Thank you

**Comment by engageub** [comment]
Closing this since this is not an issue and no further updates have made on this.

**Comment by contidz** [comment]
I can refer to honeygain how many proxies do you use to get 30gb/day

**Reply by SkullFace141** [reply]
100 proxies

**Reply by contidz** [reply]
Are you running 100 proxies for 10 accounts?

**Reply by yesyeahvh123** [reply]
@SkullFace141 Hello sir can you share me website you buy proxy? Thanks

**Comment by engageub** [comment]
@SkullFace141,
Could you please provide the region for which you used the proxies. Any recommendations for proxy provider would be appreciated. 

Thank you

**Reply by SkullFace141** [reply]
USA

**Comment by engageub** [comment]
This website similar to ebesucher. It does not need any extension. It has potential to earn about $5 per day with 100 IPs. Accepts both data center and residential proxies.
If you open 100 browsers with different proxies with PTP link you can earn it. The points are converted to payout points every sunday.
I have not received payout yet since I am still running it. If you have capacity for 100 browsers to run on your PC. Please sign up using the link here.

https://adnade.net/ptp/?user=engageub

After signing up with the link mentioned, just replace my username with your username and open in browser and leave it. Today is saturday, if you open it today, tomorrow it will be converted since every sunday points are converted.

It will soon be added to the code after payout is confirmed. 


Thank you

**Reply by contidz** [reply]
can i open multiple tabs with your tool

**Reply by engageub** [reply]
By default using with multiple proxies it creates multiple docker containers with different instances. If you wish to manually do it, you can visit the docker container to open multiple tabs in single container. You need to make sure each tab has different IP address.

Thank you

**Reply by contidz** [reply]
How to use internetcomemain to run PTP adnade.net ?
Thank you!

**Reply by contidz** [reply]
How to check points balance at adnade.net ?
Thank you!

**Reply by contidz** [reply]
I hide the tab in the system bar, will PTP work ?

**Reply by contidz** [reply]
![image](https://github.com/engageub/InternetIncome/assets/137675836/86d78927-2da3-48b7-9d32-945720600696)

**Reply by engageub** [reply]
The url in internetincome.sh for ebesucher has to be changed to relevant adanade.net.

**Reply by engageub** [reply]
Balance will be updated in your account after stopping the browser. You need to wait for about 5 minutes after stopping for the balance to be reflected.  You can visit the following link to check your balance. 

https://adnade.net/login.php?page=account&navaction=konto

Thank you

**Reply by engageub** [reply]
I guess it should work as long as the browser is working in background.
To verify you will need to run for 5 minutes and stop it. Then visit the account statement to see the balance updates at that time. Please find the account statement url.
https://adnade.net/login.php?page=account&navaction=kontoauszug

Thank you

**Reply by contidz** [reply]
can you develop more function to run this PTP url ?

**Reply by contidz** [reply]
![image](https://github.com/engageub/InternetIncome/assets/137675836/5bd8b6da-1e6c-4ebc-a88f-cb1beb937098)
i open 150tab with 150proxy after and here is my earnings after 2h

**Reply by engageub** [reply]
The script will be published once the payout is confirmed. However test version will be released for this.

Thank you

**Reply by contidz** [reply]
How long will I be able to use the trial version?
Thank you !

**Reply by engageub** [reply]
This is not a good credit score. It looks the credit score has been reduced now. It was giving about 0.4 per 20 seconds. I think it would be better if you stop them running now . More people might have started the instances or today is a weekend so traffic might have been reduced. 
For 230 seconds it shows about 0.55 which is nearly 10 times less than what it was supposed to be.

Thank you

**Reply by engageub** [reply]
There is no trial version. There are two branches in github Internet Income Page.
1. Main Branch
2. Test Branch

Main Branch is what you see in the main page. 
Test Branch Code is visible only after selecting the test branch from the main page.

Main Branch Link: https://github.com/engageub/InternetIncome
Test Branch Link:  https://github.com/engageub/InternetIncome/tree/test

Once the changes are pushed to test branch it will be updated here for adanade. 

Thank you

**Reply by contidz** [reply]
I have a suggestion:
When updating the official version you should add the updated function description.
Thank YOU

**Reply by engageub** [reply]
I did not understand what you mean by "you should add the updated function description".
Have you seen anything missing  in the past updates  when new code is added. 

Thank you

**Reply by engageub** [reply]
Did anyone sign up using my link. I do not see any referrals in my account.
Thank you

**Reply by engageub** [reply]
Points are converted to Eur, has anyone been able to reach payout threshold
Thank you

**Reply by contidz** [reply]
![image](https://github.com/engageub/InternetIncome/assets/137675836/f133f666-fef3-44da-b7e4-0756b5423bd0)
i ran 3h and stopped, i need you to add adnade.net to internet income script

**Reply by nkciam** [reply]
How much CPU and memory are required to run 100 browser Adnade? It's working fine with my VPS 4 CPU/128G Memory but only with 20 browsers is my CPU full and memory usage 10 GB. Could you share? Thank a lot!

**Reply by engageub** [reply]
@nkciam , You have given the answer already to us. If 4 cpu is 20 Browsers and 10 GB RAM. Then you will need to get the frequency of the cpu and multiply it. You will require 5 times more cpu than the existing cpu and you may not be required to  increase the  RAM.
24 CPU - 128G RAM would be fine for smooth running of the application according to you stats. But make sure you don't end up in loss while spending on the servers. 
Thank you

**Reply by hieunguyen8794** [reply]
Is it necessary to use browser anti detech? @engageub

**Reply by engageub** [reply]
@hieunguyen8794 Could you please rephrase the sentence what was mentioned. If you are talking about browser anti detect, the  website may require the browser being used in order to do surfing. You can access the browser and install any extension you wish in the docker container for adnade. Also, the restart is not provided for adnade container since the earnings have not been confirmed by anyone else. 

Thank you

**Reply by hieunguyen8794** [reply]
My idea is, if I open 100 regular Chrome profiles, each with its own IP, would that still work fine? I'm considering this approach because running it through Docker seems to consume a lot of resources.

**Reply by engageub** [reply]
@hieunguyen8794  Yes it works fine. You can assign different IP to each chrome profile and keep them running. As long as you have enough resources there should not be any problem. May I know if you are mentioning this in context of ebesucher or adnade.

**Reply by hieunguyen8794** [reply]
I'm talking about adnade

**Reply by engageub** [reply]
For adnade, firefox is being used in test version. You can limit the max cpu and memory per docker instance from properties.conf file.

**Reply by hieunguyen8794** [reply]
I don't know how to set the CPU properly, do you have any examples? I'm running on a VPS with 2 cores and 8GB of RAM, but I can only run 10 Adnade containers, and the CPU runs at 100%. So, if I run it with 48 cores, I can only run around 200 containers, right? 

There's also an issue with Linux, unlike Windows, when it's running at 80% capacity, vps can stop at any time.

**Reply by engageub** [reply]
Please note that 0.5 CPU is 50% of 1 core cpu. The maximum the container will use is 50% of 1 CPU out of 2 CPUs.
If you set the CPU to 1, it will use 1 full cpu out of 2 cpus at Max. 
Please note that the frequency of the CPU may vary with different architectures. You will need to calculate accordingly and decide how much it can run with 48 cores.
If you set the maximum limits for cpu and memory for each container, it will be under that limit and cpu will not stop.

Thank you

**Reply by engageub** [reply]
Hi All,
Adnade had an issue yesterday and they are giving 15000 credits for free. Check the news on the right side and claim them within 7 days. 
![image](https://github.com/engageub/InternetIncome/assets/103608400/916137a4-ce93-4ca3-9549-0814a57b6d61)


Thank you

**Comment by contidz** [comment]
Update adnade bro !
Thank bro !

**Reply by engageub** [reply]
It will be updated soon.
Thank you

**Reply by contidz** [reply]
Thank bro !

**Reply by xander710** [reply]
Any update on this?

**Reply by engageub** [reply]
Hi,
You can try the basic test version. Please note that the withdrawals have not been confirmed yet by anyone else.
Also, the code was not tested yet. You can have a look at this branch. It also has gaganode included. 
https://github.com/engageub/InternetIncome/tree/test


Thank you

**Reply by nkciam** [reply]
@engageub Do we need to open http://PUBLIC_VPS_IP:PORT from the adnane.txt list or docker container have already run it before? Thanks!

**Reply by engageub** [reply]
@nkciam, Its not required to open any ports on VPS. The urls are required only to verify if the firefox has opened the adnade website in the firefox/adnade container.
Please see the following link on how to access the urls in browser to see if firefox is working fine in background. 

https://github.com/engageub/InternetIncome#thinking-how-to-access-127001-urls-for-ebesucher-or-mystnodes-if-gui-is-not-available-in-linuxquestion

Also note that adnade earnings are usually reflected in your account in about 5 minutes after the containers are stopped.

**Reply by engageub** [reply]
Today is sunday and points should have been converted. Has anyone reached the withdrawal limit and made withdrawal?
Thank you

**Reply by engageub** [reply]
Payout from Adnade is confirmed and changes will be pushed to main branch soon.

**Reply by nkciam** [reply]
Thank mate for your confirm.

**Reply by engageub** [reply]
Changes for Adnade have been pushed to main branch.

**Reply by hieunguyen8794** [reply]
Adnade is almost not displaying ads anymore. Maybe something is wrong; the admins have not responded to messages.

**Reply by engageub** [reply]
Adnade does not usually display surfing/ads in the UI. The ads/surfing is done internally . Make sure credits are added to your balance. However surf ads will be added soon which gives 4X earnings. Please find the example url below. Recaptcha has to be solved before using it. It will be added similar to ebesucher in chrome. 
Make sure your IP address is not banned. 

Example surf url: https://adnade.net/view.php?user=engageub&multi=4

Thank you

**Reply by engageub** [reply]
Also, note that if you close the browser, you have to wait for atleast 10 minutes before starting it again. Else it may show that you are already running an instance on the same IP address.

**Reply by engageub** [reply]
Adnade and Ebesucher have been updated with recaptcha solver. With Adnade you can earn 4x and earnings are updated every minute in your Adnade Account Statement.
Enable chrome in Adnade and Ebesucher to use it. Please update if you have any issues.
The changes are pushed to test branch, so use the following link to run the chrome containers. 

https://github.com/engageub/InternetIncome/tree/test


Thank you

**Comment by engageub** [comment]
Hi,
Could you please read the recently updated [FAQ](https://github.com/engageub/InternetIncome?tab=readme-ov-file#-why-are-some-of-the-containers-for-the-same-application-offline) for issues related to DNS queries.
Earnings may also become low if the servers are not reachable. 

Thank you

**Comment by TradeBW** [comment]
Any idea how much resources each adnade instance/per ip takes up ?

**Reply by engageub** [reply]
Hi,
It depends on your CPU configuration. You will have to run it and calculate accordingly. 

Thank you

**Comment by TradeBW** [comment]
I was asking regarding ram/memory, how much it occupies per instance?

**Reply by engageub** [reply]
Each instance consumes approximately 350 MB of RAM, with usage occasionally reaching up to 500 MB. Auto Restart is already implemented in the code which clears up the data before starting.

**Reply by TradeBW** [reply]
Noted. Thanks
Also I have one query. If I deploy new app in different folder then does it create new tun2socks container for the new folder or it takes already avaible from the deployed folder?

**Reply by engageub** [reply]
Hi,
If you create a new folder and start the script, it will create new tun2socks containers.

Thank you

**Reply by TradeBW** [reply]
And if I update the properties file in existing folder and --start then will it work or will it fail ?
I dont want to double the count of the containers.

**Reply by engageub** [reply]
The script will not reuse the running containers if its present in a separate folder.
It will create new containers based on the proxies given.
You will have to add the apps in `properties.conf` and run from  a single folder if you wish to use the same proxies.

Thank you

**Comment by TradeBW** [comment]
Yea the problem is that I need to remove everything again just for 1 app.

**Reply by engageub** [reply]
Hi,

You will have to use `--delete` , add your new app in `properties.conf` file and then use `--start` . The backup data won't be deleted.

Thank you

**Reply by TradeBW** [reply]
Thats like setting up everything from the scratch. Anyways

f850309e99c
Error response from daemon: cannot join network of a non running container: c00e216fb60726d0790ea14f08973349c3486d578c43d507097b6ced4b885575
Error response from daemon: cannot join network of a non running container: c00e216fb60726d0790ea14f08973349c3486d578c43d507097b6ced4b885575
Error response from daemon: cannot join network of a non running container: c00e216fb60726d0790ea14f08973349c3486d578c43d507097b6ced4b885575
Error response from daemon: cannot join network of a non running container: c00e216fb60726d0790ea14f08973349c3486d578c43d507097b6ced4b885575
Error response from daemon: cannot join network of a non running container: c00e216fb60726d0790ea14f08973349c3486d578c43d507097b6ced4b885575
Error response from daemon: cannot join network of a non running container: c00e216fb60726d0790ea14f08973349c3486d578c43d507097b6ced4b885575

Anyone knows how to solve this problem ?I am not able to start exited containers. CTR doesnt show anything queued 
docker ps -a -f "status=exited" -q | xargs -r docker start


Using above command

**Reply by engageub** [reply]
Hi,

You just have to add the new app in existing properties file, use delete command and start.

Regarding the logs, it looks like tun2socks is not running.
Could you please check tun2socks logs. Make sure your proxy is in correct format.

Thank you

**Comment by TradeBW** [comment]
There has been some major issue lately with with one of the app, I suspect it is packetsdk as I found in the journalctl log.
The hardware which was smoothly managing 10 VMs is not able to manage in 2 since last 2 days, CPU spikes to 100% and load to 30K++ randomly.
I suspect traffmonetizer too because I find it in high cpu but couldnt find anything in journalctl for it.

Anyone experiencing this ?

**Comment by engageub** [comment]
Hi,

Could you please check which process is consuming more resources using `top` command.
Alternatively you can try `sudo docker stats` command to check resource usage of all containers.

Thank you

**Reply by TradeBW** [reply]
node dist/index.js - This is used by which app ?

**Reply by engageub** [reply]
This appears to be from repocket.

https://hub.docker.com/layers/repocket/repocket/latest/images/sha256-90b13dfc7f85c7cfc4dc7b4e941f255ee37a047a7768de1cfa661df50a123f65

**Comment by TradeBW** [comment]
Logs are very strange on server side. Not able to figure it out well.
Constantly changing

The server which could manage 4000 docker CT now not able to manage 2K.
This majorly looks after EA and Proxybase addition but it worked stable
for couple of days after adding them. Suddenly facing this on all the
servers so can't even judge if it is hardware issue.

Is there way to move apps to one server to another without reinstalling
them ? EA process is very tedious, nightmare to reinstall them. Like moving
data files or something ?

**Reply by engageub** [reply]
Hi,
With respect to the term `EA` , do you mean EarnApp?
You have an option to set CPU and memory limit per container in `properties.conf` file.
If you find any issue with respect to specific container, please contact the respective application team.

The script usually keeps a backup of the device nodes where ever necessary and does not delete by using `--delete` command.
You will have to copy all the folders on new host and then start the containers to retain the device Ids.

Thank you

**Comment by TradeBW** [comment]
Yes, Your understanding is correct for abbreviation.

I will try and move to see if same nodes get live for EA

Any ideal settings you have for cpu and ram per container?

**Comment by engageub** [comment]
Hi,

If you are not using apps which use browser inside the container, you may set the CPU limit to 1 and max RAM limit to 512 MB for each container. 

Thank you

**Comment by TradeBW** [comment]
Does docker read after the deployment is done? Or I need to reinstate
--start ?

**Reply by engageub** [reply]
I did not understand this query. Could you please rephrase. 
After copying the folders to new host you will have to start the containers using `--start` parameter.

Thank you

**Reply by TradeBW** [reply]
Like I change the properties.conf with cpu and memories settings so will the script will send resource limit settings to existing running CTs or do I need to do --delete and then do --start for settings to apply.

**Comment by TradeBW** [comment]
Also isn't 500MB too high for each container?
I had checked logs and calculated earlier that 8 container set uses approx
90 mb in total

**Reply by engageub** [reply]
You may adjust it accordingly but URnetwork sometimes consumes more than 400 Mb.
It is the maximum limit per container and does not mean you should have the RAM equivalent to the sum of all the containers limit.  

Thank you

**Reply by TradeBW** [reply]
So I use very limited apps, Only 10. None of them are Nodes, no mysterium no ur network, etc.

**Reply by engageub** [reply]
okay. You can adjust the limits based on what you observe in your PC. Make sure the limits are not too strict else your app may have connectivity issues. 

Thank you

**Reply by TradeBW** [reply]
packetstreamb6e57e308114f3c020dc0761e05300c199   22.32%    3.55GiB / 29.3GiB 
What a crazy memory leak. 
Found few proxybase also like this

**Reply by engageub** [reply]
There appears to be memory leak in the container. Could you please let me know how long you have been running this app.
3.55GiB is being used by a single container. 
Please contact packetstream support about the memory usage.

**Reply by TradeBW** [reply]
=== Top containers by CPU ===
packetstreamb6e57e308114f3c020dc0761e05300c149 50.19 3.611GiB /
earnappb6e57e308114f3c020dc0761e05300c149 50.16 3.61GiB /
packetstreamb6e57e308114f3c020dc0761e05300c199 50.12 3.61GiB /
traffmonb6e57e308114f3c020dc0761e05300c149 47.87 3.599GiB /
tunb6e57e308114f3c020dc0761e05300c188 47.86 3.6GiB /

Some more data. 50% CPU and 3GB+ ram Lol

Adding 100Mb per CT ram in properties file will help or there will be still risk of memory leaks ?

**Reply by TradeBW** [reply]
packetstreamb6e57e308114f3c020dc0761e05300c149 uptime: 0d 2h 44m 47s
earnappb6e57e308114f3c020dc0761e05300c149 uptime: 0d 2h 44m 53s
packetstreamb6e57e308114f3c020dc0761e05300c199 uptime: 0d 0h 15m 51s
traffmonb6e57e308114f3c020dc0761e05300c149 uptime: 0d 2h 44m 31s
tunb6e57e308114f3c020dc0761e05300c188 uptime: 0d 3h 15m 10s

**Comment by TradeBW** [comment]
Sure, let me observe again all the stats and performance. It was a hectic
day today just to make it stable.

Also do you have a peer2profit account ?

**Reply by engageub** [reply]
Hi,

I came to know that new users were unable to register. Do you have issues registering. 
You can add the proxies directly through the website using telegram. 

Thank you

**Comment by TradeBW** [comment]
I am a new user and I cannot register. any other ways to do it ?

**Reply by engageub** [reply]
They probably might have closed registration for new users and is available only for existing users.

**Reply by TradeBW** [reply]
Can we associate? I have really really super good ISP IPs.

**Reply by engageub** [reply]
I don't think there are other ways to register. You can try contacting their support team. 
You can add your proxies in bytelixir dashboard too.

**Comment by TradeBW** [comment]
Server itself was running for barely 3 days. Fresh deployment. I
couldn't check if the PS was restarted or not in the middle.

**Reply by engageub** [reply]
It is better to contact packetstream support to know the reason for high memory usage.

**Comment by TradeBW** [comment]
[INFO] Updating container: tund1a8c3998b243198dd32abd479ff197513 with
CPU=1, Memory=104MB
Error response from daemon: Cannot update container
f4c2a5f46febfa64d6e653ead96eecfcb72aa8cf87ea0618c95f8402c906162c: Memory
limit should be smaller than already set memoryswap limit, update the
memoryswap at the same time

How much is memory swap limit set for container apps any idea ? This alert
is for almost all.
I tried to set ram to 104 mb and cpu 1 core

**Reply by engageub** [reply]
As mentioned earlier, do not set strict memory limits. 
Also it is not required unless there are memory leaks in the applications.
In the following configuration have you set `MEMORY_RESERVATION` parameter also?

```
# CPU and Memory Parameters to limit for each docker container
# Uncomment the following lines if you wish to set CPU and memory limits
#MAX_MEMORY=1024m
#MEMORY_RESERVATION=
#CPU=1
```

Some environments (like Docker Desktop, certain orchestrators, or cgroups v2) may behave differently with swap—sometimes swap control isn’t even enabled.

You may have to use `--memory-swap` parameter along with `--memory` parameter in the code.

Thank you

**Comment by TradeBW** [comment]
I am currently doing it directly via docker stats. yet to explore memory &
cpu limits from your native script.

I have set 30-50% as an additional buffer in the memory limits along with
memory-swap. This is as per docker analysis of the avg per containers are
using on my infra as I do not run nodes.

I observe that it is not just memory leaks but also CPU leaks in the system
via docker containers causing kernel panics, High load - 30K++, etc.

I will observe if the earnings drop or not after this change, I certainly
see the resources are dropped very well and controlled now, while the
Network graph remains almost the same. Shall review if earnings drop is
there or not.

**Comment by engageub** [comment]
Hi,
Is your issue resolved by adding `--memory-swap` in the docker command line?
If so this will be added to the test branch along with Memory Reservation.

Thank you

**Comment by TradeBW** [comment]
Hello bro, I could change the maximum memory to desired value with --memory-swap parameter. However I am not still sure if it the issues are resolved yet. Still reviewing it.
Memory swap can by anything more than value we desire for memory, For eg if memory is 100 then memory swap can be 101

**Reply by engageub** [reply]
Hi ,

Thank you for your response. I wanted to confirm whether the following error still occurs after adding the memory swap:

```
Memory limit should be smaller than already set memoryswap limit, update the
memoryswap at the same time
```

If no Docker-related errors are present, it generally indicates that the containers have started successfully.

Thank you

**Comment by TradeBW** [comment]
Yes this process completed successfully.

Always make sure to give 1MB more to memory swap in logic or else it will
fail.

**Reply by engageub** [reply]
Thank you for your confirmation. This issue may be related to your PC or VPS configuration, as I have not encountered this error on a VPS running Ubuntu with the default memory settings.
However this parameter will be added to test branch soon.

**Comment by TradeBW** [comment]
I am testing it on Dedi Lxc containers now, Professional VPS config come
with high quality tech setup on backend so yes issues might be less in VPS.
One can set the same numbers in --memory and --memory-swap in order to
avoid swap or nil down the swap usage just like swapoff function.

If someone has NVme drives then pushing double the amount number to
in memory-swap is good thing as it will not let container die in case of
memory exhaustion

Swap off
--memory=500M --memory-swap=500M

Swap on
--memory=500M --memory-swap=(any number greater than 500) - Difference ot
Mem and Mem-Swap will be used as swap incase of memory/ram exhaustion.

I hope this explains and helps. Thanks

**Comment by TradeBW** [comment]
Hello any recommendations on how to clear the diskspace from the VPS/VMs ? 
Which folders are safe to periodic clean up inside docker containers ?

**Comment by engageub** [comment]
Hi,

The disk space is usually consumed due to logging of the applications.
It can also be consumed by stale docker images.
Please find the command below to clear stopped containers and stale docker images. 

```
sudo docker system prune -a
```

Automatic log rotation for Docker is already implemented in the code when logging is enabled through the properties.conf file.
If a specific container continues to generate logs indefinitely in another location, the issue should be escalated to the respective application team for further investigation

https://github.com/engageub/InternetIncome/wiki/Frequently-Asked-Questions#-how-to-clean-up-all-docker-containers-on-the-host

Thank you

**Comment by TradeBW** [comment]
I believe logs can be safely cleared ?

Any safe way to do it ?

**Reply by engageub** [reply]
Hi,

Could you please let us know which logs you were referring to.

Thank you

**Comment by TradeBW** [comment]
--- /var/lib/docker/overlay2/b1e84043991680ae2a2106400dc78e78bde1d989429ec5f185946cc89cb33547/diff/tmp/_MEIPMh9V3 --------------------------------------------------------------------------------
                                          /..
    8.6 MiB [###########################]  libpython3.12.so
    5.1 MiB [###############            ]  libcrypto.so.3
    2.6 MiB [########                   ] /cryptography
    1.9 MiB [#####                      ] /lib-dynload
    1.3 MiB [###                        ]  base_library.zip
  976.0 KiB [##                         ] /zstandard
  744.0 KiB [##                         ]  libzstd.so.1
  684.0 KiB [##                         ]  libssl.so.3
  288.0 KiB [                           ] /certifi
  200.0 KiB [                           ]  liblzma.so.5
  192.0 KiB [                           ]  _cffi_backend.cpython-312-x86_64-linux-gnu.so
  180.0 KiB [                           ]  libgcc_s.so.1
  172.0 KiB [                           ]  libexpat.so.1
  136.0 KiB [                           ] /ossl-modules
  112.0 KiB [                           ]  libz.so.1
   80.0 KiB [                           ]  libbz2.so.1.0
   52.0 KiB [                           ] /bcrypt
   48.0 KiB [                           ]  libffi.so.8
   24.0 KiB [                           ] /cryptography-41.0.7.dist-info
   
   Replicated folders exactly like this for each containers in overlay. are these sensitive ?

**Reply by engageub** [reply]
Hi,
You will have to contact the respective app team or research on your own.
There may be dependencies required by the app. 

Thank you

**Comment by TradeBW** [comment]
Yea gotta figure it out.

Also Just an update, I am not 100% sure yet, since the implementation of
the RAM and CPU limits I observe that traffmon earnings have significantly
dropped, Though stats on server are within limits and not even close to
thresholds set. It is very uncertain how the earnings have dropped without
any errors.
I have dropped a message to the Traffmon team if there have been any
changes in the app.

Also is it safe to stop & start the memory leaks containers? Maybe they
were the high earnings ones with high resource utilization but now there is
a sudden drop since I implemented monitoring & fixing of those containers
at server level.

**Comment by engageub** [comment]
Hi,
The first priority would be to follow up with the respective application team and ask them to resolve the memory leak issue. 
If there are memory leaks in the containers, you can simply restart the containers.
This will have problem if the applications have rate limits.
Some of the applications may consider that you have connected two devices at the same time after restarting immediately. 
You will have to give a delay before starting the application. 

Thank you
